### PR TITLE
QE: Add openSUSE 15.6 support

### PR DESCRIPTION
**Pull request by vandabarata**
Related to https://github.com/SUSE/spacewalk/issues/23848

### CODE REVIEW

**Comment by vandabarata** [comment]
The failed checks aren't related to my changes so that probably isn't anything to worry about

File: terracumber_config/tf_files/SUSEManager-4.3-PRV.tf [hunk 1]
```diff
@@ -104,7 +104,7 @@ module "cucumber_testsuite" {
   cc_username = var.SCC_USER
   cc_password = var.SCC_PASSWORD
 
-  images = ["centos7o", "opensuse155o", "sles15sp4o", "ubuntu2204o"]
+  images = ["centos7o", "opensuse155o", "opensuse156o", "sles15sp4o", "ubuntu2204o"]
```
Comment (Bischoff) [inline terracumber_config/tf_files/SUSEManager-4.3-PRV.tf:107]: not for CI (yet)

File: terracumber_config/tf_files/SUSE-Manager-4.3-PR-tests-env5.tf [hunk 2]
```diff
@@ -161,7 +161,7 @@ module "cucumber_testsuite" {
   mirror      = "minima-mirror-ci-bv.mgr.prv.suse.net"
   use_mirror_images = true
 
-  images = ["rocky9o", "opensuse155o", "sles15sp4o", "ubuntu2204o"]
+  images = ["rocky9o", "opensuse155o", "opensuse156o", "sles15sp4o", "ubuntu2204o"]
```
Comment (Bischoff) [inline terracumber_config/tf_files/SUSE-Manager-4.3-PR-tests-env5.tf:164]: I don't know about tests on PRs. Ask Jordi.

File: terracumber_config/tf_files/SUSEManager-4.3-NUE.tf [hunk 3]
```diff
@@ -102,7 +102,7 @@ module "cucumber_testsuite" {
   cc_username = var.SCC_USER
   cc_password = var.SCC_PASSWORD
 
-  images = ["centos7o", "opensuse155o", "sles15sp4o", "ubuntu2204o"]
+  images = ["centos7o", "opensuse155o", "opensuse156o", "sles15sp4o", "ubuntu2204o"]
```
Comment (Bischoff) [inline terracumber_config/tf_files/SUSEManager-4.3-NUE.tf:105]: not for CI (yet)

File: terracumber_config/tf_files/SUSEManager-4.3-SLE-update.tf [hunk 4]
```diff
@@ -96,7 +96,7 @@ module "base" {
   name_prefix = "suma-su-43-"
   use_avahi   = false
   domain      = "mgr.prv.suse.net"
-  images      = [ "sles15sp4o", "opensuse155o" ]
+  images      = [ "sles15sp4o", "opensuse155o", "opensuse156o" ]
```
Comment (Bischoff) [inline terracumber_config/tf_files/SUSEManager-4.3-SLE-update.tf:99]: I don't think so

File: terracumber_config/tf_files/SUSEManager-4.3-refenv-NUE.tf [hunk 5]
```diff
@@ -106,7 +106,7 @@ module "cucumber_testsuite" {
   cc_username = var.SCC_USER
   cc_password = var.SCC_PASSWORD
 
-  images = ["centos7o", "opensuse155o", "sles15sp4o", "ubuntu2204o"]
+  images = ["centos7o", "opensuse155o", "opensuse156o", "sles15sp4o", "ubuntu2204o"]
```
Comment (Bischoff) [inline terracumber_config/tf_files/SUSEManager-4.3-refenv-NUE.tf:109]: probably not

File: terracumber_config/tf_files/SUSEManager-Head-NUE.tf [hunk 6]
```diff
@@ -102,7 +102,7 @@ module "cucumber_testsuite" {
   cc_username = var.SCC_USER
   cc_password = var.SCC_PASSWORD
 
-  images = ["rocky8o", "opensuse155o", "ubuntu2204o", "sles15sp4o", "slemicro55o"]
+  images = ["rocky8o", "opensuse155o", "opensuse156o", "ubuntu2204o", "sles15sp4o", "slemicro55o"]
```
Comment (Bischoff) [inline terracumber_config/tf_files/SUSEManager-Head-NUE.tf:105]: not for CI

File: terracumber_config/tf_files/Uyuni-Master-K3S.tf [hunk 7]
```diff
@@ -102,7 +102,7 @@ module "cucumber_testsuite" {
   cc_username = var.SCC_USER
   cc_password = var.SCC_PASSWORD
 
-  images = ["rocky8o", "opensuse155o", "leapmicro55o", "ubuntu2204o", "sles15sp4o"]
+  images = ["rocky8o", "opensuse155o", "opensuse156o", "leapmicro55o", "ubuntu2204o", "sles15sp4o"]
```
Comment (Bischoff) [inline terracumber_config/tf_files/Uyuni-Master-K3S.tf:105]: not for CI

File: terracumber_config/tf_files/SUSE-Manager-4.3-PR-tests-env5.tf (hunk 2 above)
Comment (srbarrios) [inline terracumber_config/tf_files/SUSE-Manager-4.3-PR-tests-env5.tf:164]: We need to change all the changes matching `opensuse156o` by `opensuse156armo`

File: terracumber_config/tf_files/Uyuni-Master-PRV.tf [hunk 8]
```diff
@@ -102,7 +102,7 @@ module "cucumber_testsuite" {
   cc_username = var.SCC_USER
   cc_password = var.SCC_PASSWORD
 
-  images = ["rocky8o", "opensuse155o", "ubuntu2204o", "sles15sp4o"]
+  images = ["rocky8o", "opensuse155o", "opensuse156o", "ubuntu2204o", "sles15sp4o"]
```
Comment (Bischoff) [inline terracumber_config/tf_files/Uyuni-Master-PRV.tf:105]: not for CI

File: terracumber_config/tf_files/Uyuni-Master-podman.tf [hunk 9]
```diff
@@ -102,7 +102,7 @@ module "cucumber_testsuite" {
   cc_username = var.SCC_USER
   cc_password = var.SCC_PASSWORD
 
-  images = ["rocky8o", "opensuse155o", "leapmicro55o", "ubuntu2204o", "sles15sp4o"]
+  images = ["rocky8o", "opensuse155o", "opensuse156o", "leapmicro55o", "ubuntu2204o", "sles15sp4o"]
```
Comment (Bischoff) [inline terracumber_config/tf_files/Uyuni-Master-podman.tf:105]: not for CI (yet)

File: terracumber_config/tf_files/Uyuni-Master-refenv-NUE.tf [hunk 10]
```diff
@@ -106,7 +106,7 @@ module "cucumber_testsuite" {
   cc_username = var.SCC_USER
   cc_password = var.SCC_PASSWORD
 
-  images = ["rocky8o", "opensuse155o", "ubuntu2204o", "sles15sp4o"]
+  images = ["rocky8o", "opensuse155o", "opensuse156o", "ubuntu2204o", "sles15sp4o"]
```
Comment (Bischoff) [inline terracumber_config/tf_files/Uyuni-Master-refenv-NUE.tf:109]: I don't think so

File: terracumber_config/tf_files/Uyuni-Master-tests-code-coverage.tf [hunk 11]
```diff
@@ -102,7 +102,7 @@ module "cucumber_testsuite" {
   cc_username = var.SCC_USER
   cc_password = var.SCC_PASSWORD
 
-  images = ["rocky8o", "opensuse155o", "ubuntu2204o", "sles15sp4o"]
+  images = ["rocky8o", "opensuse155o", "opensuse156o", "ubuntu2204o", "sles15sp4o"]
```
Comment (Bischoff) [inline terracumber_config/tf_files/Uyuni-Master-tests-code-coverage.tf:105]: Probably not

File: terracumber_config/tf_files/tfvars/PR-testing-manager43.tfvars [hunk 12]
```diff
@@ -1,7 +1,7 @@
 ############ Spacewalk unique variables ############
 
 IMAGE                  = "sles15sp4o"
-IMAGES                 = ["rocky9o", "opensuse155o", "sles15sp4o", "ubuntu2204o"]
+IMAGES                 = ["rocky9o", "opensuse155o", "opensuse156o", "sles15sp4o", "ubuntu2204o"]
```
Comment (Bischoff) [inline terracumber_config/tf_files/tfvars/PR-testing-manager43.tfvars:4]: ask Jordi, but probably not

File: terracumber_config/tf_files/SUSE-Manager-4.3-PR-tests-env6.tf [hunk 13]
```diff
@@ -161,7 +161,7 @@ module "cucumber_testsuite" {
   mirror      = "minima-mirror-ci-bv.mgr.prv.suse.net"
   use_mirror_images = true
 
-  images = ["rocky9o", "opensuse155o", "sles15sp4o", "ubuntu2204o"]
+  images = ["rocky9o", "opensuse155o", "opensuse156o", "sles15sp4o", "ubuntu2204o"]
```
Comment (Bischoff) [inline terracumber_config/tf_files/SUSE-Manager-4.3-PR-tests-env6.tf:164]: same

**Review by Bischoff — CHANGES_REQUESTED** [review]
see remarks

File: terracumber_config/tf_files/SUSEManager-5.0-build-validation-NUE.tf [hunk 14]
```diff
@@ -645,6 +645,33 @@ module "opensuse155arm-minion" {
...
+  }
+  server_configuration = {
+    hostname = "suma-bv-50-srv.mgr.suse.de"
+  }
+  auto_connect_to_master  = false
+  use_os_released_updates = false
+  ssh_key_path            = "./salt/controller/id_rsa.pub"
+
+  additional_packages = [ "venv-salt-minion" ]
+  install_salt_bundle = true
```
[the remark this answers is not in the document]
Comment (nodeg) [reply terracumber_config/tf_files/SUSEManager-5.0-build-validation-NUE.tf:672]: It is fine to leave it in. We take care of the issues with a Salt cleanup in the test suite.

File: terracumber_config/tf_files/SUSEManager-4.3-SLE-update.tf [hunk 15]
```diff
@@ -96,7 +96,7 @@ module "base" {
   name_prefix = "suma-su-43-"
   use_avahi   = false
   domain      = "mgr.prv.suse.net"
-  images      = [ "sles15sp4o", "opensuse155o" ]
+  images      = [ "sles15sp4o", "opensuse155o"]
```
[the remark this answers is not in the document]
Comment (nodeg) [reply terracumber_config/tf_files/SUSEManager-4.3-SLE-update.tf:99]: I think this is a mistake and the whitespace at the end should be there. It is intentional, like after the beginning bracket.

File: terracumber_config/tf_files/SUSEManager-4.3-build-validation-NUE.tf [hunk 16]
```diff
@@ -115,7 +115,7 @@ module "base_core" {
   name_prefix = "suma-bv-43-"
   use_avahi   = false
   domain      = "mgr.suse.de"
-  images      = [ "sles12sp5o", "sles15sp1o", "sles15sp2o", "sles15sp3o", "sles15sp4o", "sles15sp5o", "slemicro51-ign", "slemicro52-ign", "slemicro53-ign", "slemicro54-ign", "slemicro55o", "almalinux8o", "almalinux9o", "centos7o", "libertylinux9o", "oraclelinux9o", "rocky8o", "rocky9o", "ubuntu2004o", "ubuntu2204o", "debian10o", "debian11o", "debian12o", "opensuse155o" ]
+  images      = [ "sles12sp5o", "sles15sp1o", "sles15sp2o", "sles15sp3o", "sles15sp4o", "sles15sp5o", "slemicro51-ign", "slemicro52-ign", "slemicro53-ign", "slemicro54-ign", "slemicro55o", "almalinux8o", "almalinux9o", "centos7o", "libertylinux9o", "oraclelinux9o", "rocky8o", "rocky9o", "ubuntu2004o", "ubuntu2204o", "debian10o", "debian11o", "debian12o", "opensuse155o", "opensuse156o"]
```
[the remark this answers is not in the document]
Comment (nodeg) [reply terracumber_config/tf_files/SUSEManager-4.3-build-validation-NUE.tf:118]: I think this is a mistake and the whitespace at the end should be there. It is intentional, like after the beginning bracket.

**Review by nodeg — CHANGES_REQUESTED** [review]
See remarks. Furthermore, adjustment in the `main.tf` file for the Uyuni BV in NUE are missing :wink:

**Review by Bischoff — APPROVED** [review]
I did not check the MACs. The rest looks good to me.

**Review by nodeg — APPROVED** [review]
LGTM overall. Thank you, Vanda!

**Pull request merged**